src/codegen/ruby/def/resource.rb

```diff
@@ -28,25 +30,43 @@ def write(package)
       ) + imports).each do |import|
         f.puts "import #{import};"
       end
-      
-      f.puts
+
       f.puts "@Value.Immutable"
       f.puts "@JsonDeserialize(as=Immutable#{name}.class)"
       f.puts "public interface #{name} {"
-      
-      fields.each { |field| field.write(f) }
-      
+
+      query_string_method = '    return new com.github.princesslana.eriscasper.data.util.QueryStringBuilder()'
+
+      fields.each { |field|
+        field.write(f)
+        if query_string
+          query_string_method += "\n      .add#{field.raw_java_type}(\"#{field.property_name}\", #{field.java_name}())"
+        end
+      }
+
+      if query_string
+        f.puts '  default String toQueryString() {'
+        f.puts query_string_method
```

Comment:
Instead of building up the query string here, can we have `Data.toQueryString` (to match `Data.toJsonString`)?

This could be implemented by having Jackson convert the object to a Map, and then converting that map to the query string

Reply:
Sounds like a good idea, wasn't sure a good way to implement this. It might be smarter to implement our own Field parser for stuff like that, just so we don't need to parse it twice if you understand what I'm saying. It really shouldn't be hard to make. I do however like that more than the ruby solution. Just wanted to keep it consistent with .Java for now until we find a better solution (which it looks like we have).

Reply:
Ended up using jackson to turn it into a json node then mapped it out. Since jackson is nicely built up with parsing everything we have thought it was a fruitless endeavor to not use it and build our own.